packages/country-config/src/forms.ts

```diff
@@ -1,18 +1,89 @@
 // @TODO: Yet to be implemented! Placeholders for NPM publish.
 
 /**
- * E-Signet popup button form definition
+ * E-Signet REDIRECT button form definition.  Calls E-Signet /authorize
  */
-export const popupButton = ({
-  /** URL to OpenCRVS-MOSIP gateway (e.g. https://opencrvs-mosip-gateway.farajaland.opencrvs.org) */
-  url,
-}: {
-  url: string;
-}) => {
+
+
+export const returnRedirectFormField = (esignetAuthUrl: string, openIdProviderClientId: string, openIdProviderClaims: string) => {
+  // `${nidSystemSetting?.openIdProviderBaseUrl}authorize`
+  const url = new URL(esignetAuthUrl)
+
+  url.searchParams.append(
+    'client_id',
+    openIdProviderClientId || ''
+  )
+  url.searchParams.append('response_type', 'code')
+  url.searchParams.append('scope', 'openid profile')
+  url.searchParams.append('acr_values', 'mosip:idp:acr:static-code')
+  url.searchParams.append('claims', openIdProviderClaims || '')
+
+  /*
+
+  TODO: Understand from Tahmid about how to handle this:
```

Comment:
@tahmidrahman-dsi can you advise how we can send this state and callback URL to E-Signet using the REDIRECT button please?

Reply:
@euanmillar Current implementation saves no state. But can be saved if the requirement needs. Before developing, it would be great if I have a bit more understanding from my side. 

```
client_id=$openIdProviderClientId
response_type='code'
scope='openid profile'
acr_values='mosip:idp:acr:static-code'
claims=$openIdProviderClaims
```

As per my understanding, the above parameters are appended as search query params to the redirect url (`esignetAuthUrl`). And when the redirection is completed, eSignet forwards user to the opencrvs form url with param `authorized=true` which causes calling the callback url `esignetUserinfoUrl`. But you need those params for calling the `esignetUserinfoUrl`. Isn't it?

If my understanding is correct, then could it done in such way, that when eSignet forwards to opencrvs form, it forwards back with those same params? Then we can trigger the callback depending on those params instead of `authorized=true` ? Please let me know your thoughts. 

cc: @naftis

Reply:
You are correct to this step:
>As per my understanding, the above parameters are appended as search query params to the redirect url (esignetAuthUrl). 

After that though, the above URL redirects back to OpenCRVS with
```
https://register.mosip.opencrvs.dev/drafts/1234-1234-1234-1234/informant?state=same-that-you-supplied&nonce=same-that-you-supplied&code=authorization_code&error_description=in_case_of_error_this_is_sent&error=error_code
```
The onmount-fetch should be able to be configured so that for example a `code` exists. To the `state` we can set anything that E-Signet just sends us back, so we could put something like "?state=trigger-onmount" there to be explicit. So yes, we can use these parameters instead of `authorized=true`. This `code` is then used to fetch the details from @opencrvs/mosip/server, which then again fetches the userinfo from E-Signet.

https://docs.esignet.io/integration/relying-party
The full flow described above is documented here.